Fix for issue #777 - Neo4j support.

Rffsilva opened the pull request:
Main changes:

- Added support for neo4j migrations.
- Included neo4j docker container.
- Changed core Flyway.java to get MetaDataTable from DBSupport.
- Needed to proxy neo4j library classes so included cglib.
- Added MigrationTestCase for neo4j and test structure with cypher.
- Changed tests to test resource structure.

Rffsilva committed:
Added support for neo4j database, added tests and test structure.
Changed travis configuration to ignore JAVA_OPTIONS

Rffsilva committed:
Changed version to 4.2.1-ScuteraTech

axelfontaine commented on flyway-core/pom.xml:
```diff
@@ -210,6 +210,29 @@
+		    <version>3.2.3</version>
+		    <scope>test</scope>
+		</dependency>
+		<dependency>
```
Please do not add any required dependencies. Flyway is a library with zero required transitive dependencies and we want it to stay that way.

Rffsilva replied:
Cleaned transitive dependencies.

axelfontaine replied:
No, they are still there...

Rffsilva replied:
I removed the neo4j-configuration and neo4j-ogm from the pom file.

Rffsilva committed:
Removed transitive dependencies.

Rffsilva committed:
Merge branch 'master' into master

Rffsilva committed:
Merge branch 'master' into master

axelfontaine commented on flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/neo4j/Neo4JMetaDataTable.java:
```diff
@@ -0,0 +1,314 @@
+ * @author Ricardo Silva (ScuteraTech)
+ *
+ */
+public class Neo4JMetaDataTable implements MetaDataTable {
```
Why the separate class?

Rffsilva replied:
Hello axel, the main reason for the separate class was that the MetaDataTableImpl has many methods with sql implementations that are not supported by neo4j since it's query language differs from sql.

joelhandwell replied:
@axelfontaine is this "separating class problem" only the block to proceed this to be merged?

Rffsilva committed:
Merge branch 'master' into master

felipebn committed:
- Removing neo4j jdbc shaded dependency
- Including neo4j jdbc bolt and http driver
- Refactoring to have distinct drivers
- including neo4j into compose
- removing testcontainers for neo4j

felipebn committed:
Merge remote-tracking branch 'origin/master' into ScuteraTech

Rffsilva committed:
Merge pull request #2 from felipebn/master

Removing shaded neo4j-jdbc dependency

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/flyway/flyway?pullRequest=1771) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/flyway/flyway?pullRequest=1771) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: felipebn<br/>:x: Rffsilva<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/flyway/flyway?pullRequest=1771) it.</sub>

Rffsilva committed:
Merge pull request #1 from felipebn/ScuteraTech

Fix for neo4j-jdbc shaded dependency

felipebn committed:
DB Cleanup fix

- Including constraints cleanup
- Removing usage of spring Assert class on test
- Including cleanup test

felipebn committed:
Fixes to work with index creation

Despite being schemaless, neo4j cannot handle changes to schema (i.e.
constraints creation) together with other queries like node/relationship
creation.

This fixes the problem, but it still not possible to create constraints
and add nodes in the same migration unit, in this cases one should split
constraint creation and graph changes in 2 files.

- Fixing some code from last commit as we should not change constraints
in the same transaction where we touch nodes and relationships
- Setting dbsupport to single connection
- Including fixes to proper migrate with constraint creation
- Removing unused connection proxy as it was replaced with connection
enhancer

felipebn committed:
Syncing with main repo changes

felipebn committed:
Merge remote-tracking branch 'origin/master' into ScuteraTech

felipebn committed:
Fixing version

felipebn committed:
SXSUITE-3070 Starting neo4j after uuid index introduction generates
exception

Fixing regex to support CREATE INDEX

felipebn committed:
Merge pull request #3 from Rffsilva/ScuteraTech

Merging fix to handle CREATE INDEX the same way we handle CREATE CONSTRAINT

felipebn committed:
Reverting username/password/url for neo4j test

michaelahlers commented:
Any news on this feature?

elR1co commented:
Any update on this ?

felipebn committed:
Back porting fix to support Postgresql 11

felipebn committed:
Merge pull request #4 from felipebn/ScuteraTech

Back porting fix to support Postgresql 11

piers-williams commented:
Thank you for your submission. We cannot merge this at this time as it would need re-implementing in the [https://github.com/flyway/flyway-community-db-support](https://github.com/orgs/red-gate/projects/71/views/flyway-community-db-support) repository.
